src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -33,34 +33,103 @@ internal interface IDiagnosticAnalyzerService
+        /// <param name="workspace">Workspace to fetch the diagnostics for.</param>
+        /// <param name="id">
+        /// Specific id to scope the returned diagnostics. This id must correspond to the
+        /// <see cref="Common.UpdatedEventArgs.Id"/> associated with <see cref="DiagnosticsUpdatedArgs"/> event.
+        /// </param>
+        /// <param name="includeSuppressedDiagnostics">Indicates if diagnostics suppressed in source via pragmas and SuppressMessageAttributes should be returned.</param>
+        /// <param name="includeNonLocalDocumentDiagnostics">
+        /// Indicates if non-local document diagnostics must be returned.
+        /// Non-local diagnostics are the ones reported by analyzers either at compilation end callback OR
+        /// in a different file from which the callback was made. Entire project must be analyzed to get the
+        /// complete set of non-local document diagnostics.
+        /// </param>
+        /// <param name="cancellationToken">Cancellation token.</param>
         Task<ImmutableArray<DiagnosticData>> GetSpecificCachedDiagnosticsAsync(Workspace workspace, object id, bool includeSuppressedDiagnostics, bool includeNonLocalDocumentDiagnostics, CancellationToken cancellationToken);
 
         /// <summary>
         /// Get diagnostics currently stored in the source. returned diagnostic might be out-of-date if solution has changed but analyzer hasn't run for the new solution.
         /// </summary>
-        Task<ImmutableArray<DiagnosticData>> GetCachedDiagnosticsAsync(Workspace workspace, ProjectId? projectId, DocumentId? documentId, bool includeSuppressedDiagnostics, bool includeNonLocalDocumentDiagnostics, CancellationToken cancellationToken);
+        /// <param name="workspace">Workspace for the document/project/solution to compute diagnostics for.</param>
+        /// <param name="projectId">Optional project to scope the returned diagnostics.</param>
+        /// <param name="documentId">Optional document to scope the returned diagnostics.</param>
+        /// <param name="includeSuppressedDiagnostics">Indicates if diagnostics suppressed in source via pragmas and SuppressMessageAttributes should be returned.</param>
+        /// <param name="includeLocalDocumentDiagnostics">
+        /// Indicates if local document diagnostics must be returned.
+        /// Local diagnostics are the ones that are reported by analyzers on the same file for which the callback was received
+        /// and hence can be computed by analyzing a single file in isolation.
+        /// </param>
+        /// <param name="includeNonLocalDocumentDiagnostics">
+        /// Indicates if non-local document diagnostics must be returned.
+        /// Non-local diagnostics are the ones reported by analyzers either at compilation end callback OR
+        /// in a different file from which the callback was made. Entire project must be analyzed to get the
+        /// complete set of non-local document diagnostics.
```

Comment:
a good reason for this to be an options object.  would just be one set of docs :)

can be done in the future.